extensions/amp-form/0.1/amp-form.js

```diff
@@ -1296,12 +1296,24 @@ export class AmpForm {
         p = this.ssrTemplateHelper_
           .applySsrOrCsrTemplate(devAssert(container), data)
           .then((rendered) => {
-            rendered.id = messageId;
-            rendered.setAttribute('i-amphtml-rendered', '');
+            // TODO(caroqliu): Simplify section appending rendered contents to DOM.
+            let renderContainer;
+            if (isArray(rendered)) {
```

Comment:
Thinking about it now, would we ever actually expect an array for form renders? I.e. is input `data` ever an array?

Reply:
Omg, no! 🤦 
https://github.com/ampproject/amphtml/blob/98d57b0ebb28b82b9c32b46e4a400d395070ecfd/extensions/amp-form/0.1/amp-form.js#L1281-L1287

I'll replace the changes on amp-form with a `devAssert` that the result is not an array.

Reply:
Yea, we're handling both array and non-array cases in a single codepath for SSR. But I still think the same invariant applies -- we should only receive an array of elements if `data` is an array.

Since `setHtml` will always return an array now, you could just assert that length == 1. 

Anyways, let's leave this for the "simplify" TODO. You can also file an issue and reference this PR thread instead of self-assigning.